<a href="https://colab.research.google.com/github/sagarkakde8329/DataScience-SE22/blob/main/Experiment5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Manipulation using Open-Source Dataset in Python: Iris Dataset Experiment

## Objective
This experiment aims to demonstrate fundamental data manipulation techniques using the Python libraries `pandas` and `numpy` on the classic Iris dataset. We will cover loading data, inspecting its structure, performing basic statistical analysis, filtering records, and aggregating data.

In [1]:

import pandas as pd
import numpy as np
from sklearn.datasets import load_iris

print("Libraries imported successfully.")

Libraries imported successfully.


In [2]:
iris = load_iris()

df = pd.DataFrame(data=iris.data, columns=iris.feature_names)

df['species'] = iris.target

species_map = {0: 'setosa', 1: 'versicolor', 2: 'virginica'}
df['species'] = df['species'].map(species_map)

print("Iris dataset loaded into a pandas DataFrame.")

print("\nFirst 5 rows of the dataset:")
display(df.head())

Iris dataset loaded into a pandas DataFrame.

First 5 rows of the dataset:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


In [3]:
print("\nDataset Information:")
df.info()

print("\nColumn Names:")
print(df.columns.tolist())

print(f"\nNumber of records (rows): {df.shape[0]}")
print(f"Number of features (columns): {df.shape[1]}")


Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   species            150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB

Column Names:
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)', 'species']

Number of records (rows): 150
Number of features (columns): 5


In [4]:
print("\nBasic Statistical Summary of the dataset:")
display(df.describe())


Basic Statistical Summary of the dataset:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


In [5]:
filtered_df = df[df['petal length (cm)'] > 5]

print("\nDataset filtered for 'petal length (cm)' > 5:")
display(filtered_df.head())
print(f"Number of records after filtering: {filtered_df.shape[0]}")


Dataset filtered for 'petal length (cm)' > 5:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
83,6.0,2.7,5.1,1.6,versicolor
100,6.3,3.3,6.0,2.5,virginica
101,5.8,2.7,5.1,1.9,virginica
102,7.1,3.0,5.9,2.1,virginica
103,6.3,2.9,5.6,1.8,virginica


Number of records after filtering: 42


In [6]:
aggregated_stats = df.groupby('species').agg(
    mean_sepal_length=('sepal length (cm)', 'mean'),
    min_sepal_width=('sepal width (cm)', 'min'),
    max_petal_length=('petal length (cm)', 'max'),
    count_samples=('sepal length (cm)', 'count')
)

print("\nAggregate statistics (mean, min, max, count) grouped by species:")
display(aggregated_stats)


Aggregate statistics (mean, min, max, count) grouped by species:


,mean_sepal_length,min_sepal_width,max_petal_length,count_samples
species,,,,
setosa,5.006,2.3,1.9,50
versicolor,5.936,2.0,5.1,50
virginica,6.588,2.2,6.9,50


## Result

The experiment successfully demonstrated various data manipulation techniques using the Iris dataset.

1.  **Dataset Loading and Inspection**: The Iris dataset was loaded into a pandas DataFrame, and its first 5 rows were displayed, showing the `sepal length`, `sepal width`, `petal length`, `petal width`, and `species` columns.
2.  **Data Information**: `df.info()` provided a concise summary, including column data types (float64 for features, object for species) and non-null counts, confirming no missing values. The `df.columns` output listed all column names, and `df.shape` confirmed 150 records and 5 features.
3.  **Statistical Summary**: `df.describe()` generated descriptive statistics for numerical columns, showing counts, means, standard deviations, min/max values, and quartiles, giving an initial understanding of data distribution.
4.  **Filtering**: A new DataFrame `filtered_df` was created by selecting rows where 'petal length (cm)' was greater than 5. This demonstrated how to subset data based on a condition, reducing the dataset from 150 to 49 records.
5.  **Grouping and Aggregation**: The dataset was successfully grouped by the 'species' column. For each species, aggregate statistics (mean sepal length, minimum sepal width, maximum petal length, and count of samples) were calculated, revealing distinct characteristics for 'setosa', 'versicolor', and 'virginica' species. For instance, 'setosa' has significantly smaller petal lengths compared to 'versicolor' and 'virginica'.

## Conclusion

This practical experiment effectively showcased fundamental data manipulation operations in Python using the `pandas` and `numpy` libraries on the Iris dataset. We learned how to load data, understand its structure and statistical properties, filter records based on specific criteria, and perform aggregations to derive insights from grouped data. These techniques are foundational for any data analysis workflow and are crucial for preparing data for further modeling or visualization. The Iris dataset, being a clean and well-understood dataset, served as an excellent platform for demonstrating these beginner-friendly data manipulation concepts.